bunq/sdk/json/anchored_object_model_adapter.py

```diff
@@ -0,0 +1,70 @@
+from bunq import AnchoredObjectInterface
+from bunq.sdk.exception.bunq_exception import BunqException
+from bunq.sdk.json import converter
+from bunq.sdk.model.generated import endpoint
+from bunq.sdk.model.generated import object_
+
+
+class AnchoredObjectModelAdapter(converter.JsonAdapter):
+    _ERROR_MODEL_NOT_FOUND = '{} is not in endpoint nor object.'
+
+    __STRING_FORMAT_UNDERSCORE = '_'
+
+    _override_field_map = {
+        'ScheduledPayment': 'SchedulePayment',
+        'ScheduledInstance': 'ScheduleInstance',
+    }
+
+    @classmethod
+    def deserialize(cls, cls_target, obj_raw):
+        """
+        :type cls_target: BunqModel
+        :type obj_raw: int|str|bool|float|list|dict|None
+
+        :rtype: T
+        """
+
+        model_ = super()._deserialize_default(cls_target, obj_raw)
+
+        if isinstance(
+                model_,
+                AnchoredObjectInterface
+        ) and model_.is_all_field_none():
+            for field in model_.__dict__:
+                object_class = cls._get_object_class(field)
+                contents = super()._deserialize_default(object_class, obj_raw)
+
+                if contents.is_all_field_none():
+                    setattr(model_, field, None)
+                else:
+                    setattr(model_, field, contents)
+
+        return model_
+
+    @classmethod
+    def can_serialize(cls):
+        return False
+
+    @classmethod
+    def _get_object_class(cls, class_name):
+        """
+        :type class_name: str
+        :rtype: BunqModel
```

Comment:
Type[BunqModel] ?

Reply:
Please see comment above regarding a similar comment you made.

Reply:
see above.